### ISSUE TITLE
setup piserver as a os selector for remote Pi's with no authentication

### ISSUE THREAD

**Issue by thecableguy007** [raspberrypi/piserver#122]
Hi all, 
Great project!
I tried it and it works great!
I can see the use of this for a classroom where you need restrictions.

I don't know if this question/request has already been made?

Is there a way to use this Piserver as an OS selector and have full rights on the remote pi as if you where running directly off a SD?

The Idea of using a Piserver and not have to write a sd all the time, just select the image you want then reboot the remote pi,

This could be used for testing and if you scrap an OS image, you just replace it.

You could see this like the Piserver would be a multi os boot selector.

This could be used for testing and if you scrap an OS image, you just replace it.

I also have a bunch of displays at my job that are already used for signage with pi's and every time I make changes, I have to go and swap out that sd card. I would just need to put the new image on the server, point the pi's mac address to the new image  

Thanks

David

**Comment by maxnet** [comment]
>Is there a way to use this Piserver as an OS selector and have full rights on the remote pi as if you where running directly off a SD?

Yes, but not for "hostile" environments like classrooms. And I do not recommend it for your workplace either.

See: https://github.com/raspberrypi/piserver/issues/114#issuecomment-716166719 for configuring NFS to serve read-write.

This means anyone that has access to your local network can access and write to any file.
It is possible to configure NFS to restrict by IP, but that is also not very secure.
In a classroom all it would take for a student to get access to everyone's files, would be to sneak his own RPi OS SD card into the Pi, login as root locally, and mount the NFS share.
That Pi would receive the same IP as when network booting based on MAC, and would be fully trusted by server.


A more secure variant would be to use iSCSI -which supports password authentication- instead of NFS.
But then you have to store the password somewhere. And then you may still end up using a SD card for that...
(For demonstrative purposes you can try the iSCSI support offered by Berryboot: https://www.berryterminal.com/doku.php/storing_your_files_on_a_synology_nas_using_iscsi
But you can also achieve the same manually by creating an initramfs on RPI OS).
May be possible to (ab)use the Pi 4 EEPROM for iSCSI password storage instead. But no implementation that uses that exists so far as far as I am aware.
